core/modules/installer/installer.browser.inc

```diff
@@ -0,0 +1,470 @@
+  }
+
+  $projects = installer_browser_get_installed_projects();
+  $missing = installer_browser_get_missing_dependencies($projects);
+  // If there are missing dependencies, go to install dependencies.
+  if (count($missing) > 0) {
+    backdrop_goto('admin/modules/installer-browser/install/install_dependencies');
+  }
+  else {
+    installer_browser_get_destination_after_install();
+  }
+}
+
+/**
+ * Determines the types of installed projects.
+ */
+function installer_browser_get_installed_types() {
+  $types = array();
+  $projects = installer_browser_get_installed_projects();
+  foreach ($projects as $project) {
+    $types[$project['type']] = $project['type'];
+  }
+
+  return $types;
+}
+
+
+
+/**
+ * Determines the form destination after installed projects.
+ */
+function installer_browser_get_destination_after_install() {
+  $types = installer_browser_get_installed_types();
+
+  $layout = in_array('layout', $types);
+  $theme = in_array('theme', $types);
+  $module = in_array('module', $types);
+
+  $layout_message = t('One or more layouts were installed. You may enable layouts on the !link page.', array('!link' => l(t('Layout'), 'admin/structure/layouts')));
+
+  if ($layout) {
+    // Installed a layout and other project(s), continue the wizard. There's no
+    // step for layouts in the wizard, since there's nothing we could do there.
+    if ($theme || $module) {
+      if (!$theme) {
+        backdrop_set_message($layout_message);
+        backdrop_goto('admin/modules/installer-browser/install/enable');
+      }
+      else {
+        backdrop_set_message($layout_message);
+        backdrop_goto('admin/modules/installer-browser/install/theme');
+      }
+    }
+    // Installed only a layout, go straight to layout list.
+    else {
+      backdrop_set_message($layout_message);
+      backdrop_goto('admin/structure/layouts');
+    }
+  }
+  // Theme and module, go to theme stage of wizard.
+  elseif ($theme && $module) {
+    backdrop_goto('admin/modules/installer-browser/install/theme');
+  }
+  // Only a theme, go to theme list.
+  elseif ($theme && !$module) {
+    backdrop_set_message(t('One or more themes were downloaded. You may enable them now.'));
+    backdrop_goto('admin/appearance');
+  }
+  // Only a module, jump to the module enable stage of the wizard.
+  else {
+    backdrop_goto('admin/modules/installer-browser/install/enable');
+  }
+}
+
+/**
+ * Gets the dependencies for installed projects.
+ */
+function installer_browser_get_missing_dependencies($projects) {
+  $modules = system_rebuild_module_data();
+
+  $missing = array();
+
+  foreach ($projects as $project) {
+    if ($project['type'] == 'module') {
+      $dependencies = array();
+      if (isset($modules[$project['name']])) {
+        foreach ($modules[$project['name']]->info['dependencies'] as $dependency) {
+          if (!isset($modules[$dependency])) {
+            $dependencies[] = $dependency;
```

Comment:
This function doesn't distinguish between module dependencies and project dependencies. e.g.:

```
dependencies[] = project_name:sub_module
```

Which would download `project_name` project but enable `sub_module.module`.

This is a new(ish) feature added in the last year, see https://www.drupal.org/node/542202#dependencies. We support this in Backdrop 1.2.4+ as well (https://github.com/backdrop/backdrop-issues/issues/1331).